llvm/test/CodeGen/MIR/Generic/scalable-vector-type.mir

```diff
@@ -0,0 +1,23 @@
+# RUN: llc -run-pass=none -o - %s | FileCheck %s
+
+---
+name:            scalable_vector_type_s
+body: |
+  bb.0:
+    ; CHECK-LABEL: name: scalable_vector_type_s
+    ; CHECK: [[DEF:%[0-9]+]]:_(<vscale x 1 x s8>) = IMPLICIT_DEF
+    ; CHECK-NEXT: [[COPY:%[0-9]+]]:_(<vscale x 1 x s8>) = COPY [[DEF]](<vscale x 1 x s8>)
+    %0:_(<vscale x 1 x s8>) = IMPLICIT_DEF
+    %1:_(<vscale x 1 x s8>) = COPY %0
+...
+
+---
+name:            scalable_vector_type_p
+body: |
+  bb.0:
+    ; CHECK-LABEL: name: scalable_vector_type_p
+    ; CHECK: [[DEF:%[0-9]+]]:_(<vscale x 1 x p0>) = IMPLICIT_DEF
+    ; CHECK-NEXT: [[COPY:%[0-9]+]]:_(<vscale x 1 x p0>) = COPY [[DEF]](<vscale x 1 x p0>)
+    %0:_(<vscale x 1 x p0>) = IMPLICIT_DEF
+    %1:_(<vscale x 1 x p0>) = COPY %0
+...
```

Comment:
should also have some tests with different numbers of elements and pointer addrspaces

Reply:
Also needs testing for the parser error cases

Reply:
And some real world examples for RISC-V and AArch64, e.g, `<vscale x 4 x s32>` .

Reply:
I'm not sure what making this target specific would do for a type like `<vscale x 4 x s32>`. I think we achieve the same thing by testing it under Generic folder. Did you have anything in mind that you'd like to see that needs to be target specific

Reply:
No worries. Thanks for doing this.